Alerting Notification Policy: Support org-specific provisioning

Cleymax opened the pull request:
Adds the `org_id` attribute to be consistent with other org-specific resources

Cleymax committed:
Alerting Notification Policy: Support org-specific provisioning

Cleymax committed:
Follow var name, add check read error

Cleymax committed:
Add unit tests for Alerting Notification Policy Org

Cleymax committed:
Update documentation for notification policy

julienduchesne commented on internal/resources/grafana/resource_alerting_notification_policy.go:
```diff
@@ -179,42 +184,53 @@ func policySchema(depth uint) *schema.Resource {
 	resp, err := client.Provisioning.GetPolicyTree()
-	if err != nil {
-		return diag.FromErr(err)
+	if err, shouldReturn := common.CheckReadError("notification policy", data, err); shouldReturn {
```
Unneeded for notification policy, I think. It should never return 404

Cleymax replied:
Yes, there will always be a defined receiver.

julienduchesne commented on internal/resources/grafana/resource_alerting_notification_policy_test.go:
```diff
@@ -139,6 +139,57 @@ func TestAccNotificationPolicy_disableProvenance(t *testing.T) {
+func testAccNotificationPolicyInOrg() string {
+	return fmt.Sprintf(`
+	resource "grafana_organization" "test" {
+		name = "test"
```
Template this name with a random string

julienduchesne commented on internal/resources/grafana/resource_alerting_notification_policy_test.go:
```diff
@@ -139,6 +139,57 @@ func TestAccNotificationPolicy_disableProvenance(t *testing.T) {
+					alertingNotificationPolicyCheckExists.exists("grafana_notification_policy.test", &policy),
+					checkResourceIsInOrg("grafana_notification_policy.test", "grafana_organization.test"),
+				),
+			},
```
Add another step to check that destroying within the org works. See other resources for example

julienduchesne commented on internal/resources/grafana/resource_alerting_notification_policy_test.go:
```diff
@@ -139,6 +139,57 @@ func TestAccNotificationPolicy_disableProvenance(t *testing.T) {
+		ProviderFactories: testutils.ProviderFactories,
+		CheckDestroy: resource.ComposeTestCheckFunc(
+			orgCheckExists.destroyed(&org, nil),
+			alertingNotificationPolicyCheckExists.destroyed(&policy, &org),
```
Unneeded, the org is destroyed so of course the policy will not be queryable

Cleymax committed:
Fix tests, policy error

julienduchesne committed:
Fix some issues:
- API returns 404 when it's not ready
- Add destroy check in test
- Add missing contact point in test

julienduchesne committed:
Remove trailing whiteline

julienduchesne commented:
Thanks!
